example_function_test.go

```diff
@@ -0,0 +1,26 @@
+package logrus_test
+
+import (
+	"fmt"
+	log "github.com/sirupsen/logrus"
+	"testing"
+)
+
+func TestLogger_LogFn(t *testing.T) {
+	log.SetFormatter(&log.JSONFormatter{})
+	log.SetLevel(log.WarnLevel)
+
+	log.InfoFn(func() []interface{} {
+		fmt.Println("This is never run")
```

Comment:
Can we improve the test to actually ensure that the function is indeed not called ?